frontend/src/components/auth/student-login/NameSelection.tsx

```diff
@@ -1,105 +1,97 @@
-import React, { useContext } from "react";
-import { Controller, useForm } from "react-hook-form";
+import React, { useContext, useState } from "react";
 import { useHistory } from "react-router-dom";
+import { useQuery } from "@apollo/client";
 import {
   Box,
   FormControl,
   FormErrorMessage,
   FormLabel,
 } from "@chakra-ui/react";
-import { Select } from "chakra-react-select";
+import { OptionBase, Select, SingleValue } from "chakra-react-select";
 
+import GET_CLASS_BY_TEST_SESSION from "../../../APIClients/queries/ClassQueries";
+import { StudentResponse } from "../../../APIClients/types/ClassClientTypes";
 import { STUDENT_SIGNUP_IMAGE } from "../../../assets/images";
 import { ASSESSMENT_SUMMARY_PAGE, HOME_PAGE } from "../../../constants/Routes";
 import AuthContext from "../../../contexts/AuthContext";
 import AuthWrapper from "../AuthWrapper";
 import NavigationButtons from "../teacher-signup/NavigationButtons";
 
-const students = [
-  {
-    label: "Evans",
-    value: "Evans",
-  },
-  {
-    label: "Tony",
-    value: "Tony",
-  },
-  {
-    label: "John",
-    value: "John",
-  },
-  {
-    label: "Enzo",
-    value: "Enzo",
-  },
-  {
-    label: "Julia",
-    value: "Julia",
-  },
-  {
-    label: "Emily",
-    value: "Emily",
-  },
-  {
-    label: "Afsana",
-    value: "Afsana",
-  },
-  {
-    label: "George",
-    value: "George",
-  },
-  {
-    label: "Cyrus",
-    value: "Cyrus",
-  },
-];
-const NameSelection = (): React.ReactElement => {
+interface NameSelectionProps {
+  testId: string;
+  testSessionId: string;
+}
+
+interface StudentOption extends OptionBase {
+  value: StudentResponse;
+  label: string;
+}
+
+const NameSelection = ({
+  testId,
```

Comment:
the `testId` value is currently unused, since we don't need it here. i was thinking that i would pass it via props to the `AssessmentSummaryPage` where it could query for test data and then store the results in a context that wraps the `StudentDashboard`? and then, `AuthContext` only stores info related to the student specifically.